src/DataSource.js

```diff
@@ -115,15 +117,31 @@ prototype.synchronize = function() {
   return this;
 };
 
-prototype.listener = function() { 
+prototype.listener = function() {
   return DataSourceListener(this).addListener(this._inputNode);
 };
 
+prototype.getIndex = function(field) {
+  var data = this._data,
+      index, i, len, value;
+  if (!this._indexes[field]) {
+    index = {};
+    this._indexes[field] = index;
+    this._indexFields.push(field);
+    for (i = 0, len = data.length; i < len; i++) {
+      value = this._data[i][field];
```

Comment:
What if the field includes a nested reference (`foo.bar`)? Do we want to support that? If so, you can use datalib (e.g., `dl.$(...)`) to parse the reference and create an accessor function.